solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1347 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.6.12;
+import "../../ConverterVersion.sol";
+import "../../interfaces/IConverter.sol";
+import "../../interfaces/IConverterAnchor.sol";
+import "../../interfaces/IConverterUpgrader.sol";
+import "../../../token/interfaces/IDSToken.sol";
+import "../../../utility/ContractRegistryClient.sol";
+import "../../../utility/ReentrancyGuard.sol";
+import "../../../utility/SafeMath.sol";
+import "../../../utility/TokenHandler.sol";
+import "../../../utility/TokenHolder.sol";
+import "../../../utility/Math.sol";
+
+/**
+  * @dev Liquidity Pool v3 Converter
+  *
+  * The liquidity pool v3 converter is a specialized version of a converter that manages
+  * a classic bancor liquidity pool.
+  *
+  * Even though pools can have many reserves, the standard pool configuration
+  * is 2 reserves with 50%/50% weights.
+*/
+contract LiquidityPoolV3Converter is ConverterVersion, IConverter, TokenHandler, TokenHolder, ContractRegistryClient, ReentrancyGuard {
+    using SafeMath for uint256;
+    using Math for *;
+
+    IERC20Token private constant ETH_RESERVE_ADDRESS = IERC20Token(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+    uint256 private constant MAX_UINT128 = 0xffffffffffffffffffffffffffffffff;
+    uint256 private constant MAX_UINT112 = 0xffffffffffffffffffffffffffff;
+    uint256 private constant AVERAGE_RATE_PERIOD = 10 minutes;
+    uint32 private constant PPM_RESOLUTION = 1000000;
+
+    uint256 public reserveBalances;
+    IERC20Token[] public reserveTokens;
+    mapping (IERC20Token => uint256) public reserveIds;
```

Comment:
technically, we could always sort the reserves by their addresses and have the lower value in the first 128 bits and the higher value in the 128 bits - this might reduce more SLOADs. We'll still need to read from storage in order to know if a reserve is valid or not - what do you think?

Reply:
I don't see a single place in the code where this is useful.